packages/protocol/test/common/feeburner.ts

```diff
@@ -0,0 +1,478 @@
+      ['0x0000000000000000000000000000000000000000', uniswap.address]
+    )
+  })
+
+  describe('#transfer()', () => {
+    beforeEach(async () => {
+      await tokenA.mint(feeBurner.address, new BigNumber(1e18))
+    })
+
+    it('Only owner can take tokens out', async () => {
+      await assertRevert(
+        feeBurner.transfer(tokenA.address, user, new BigNumber(1e18), { from: user })
+      )
+    })
+
+    it('Can take funds out', async () => {
+      await feeBurner.transfer(tokenA.address, user, new BigNumber(1e18))
+      assertEqualBN(await tokenA.balanceOf(user), new BigNumber(1e18))
+    })
+  })
+
+  describe('#initialize()', () => {
+    it('set the right parameters', async () => {
+      assertEqualBN(await feeBurner.maxSlippage(stableToken.address), toFixed(1))
+      assertEqualBN(await feeBurner.maxSlippage(tokenA.address), toFixed(1))
+      assertEqualBN(await feeBurner.dailyBurnLimit(stableToken.address), new BigNumber(1000000e18))
+      assertEqualBN(await feeBurner.dailyBurnLimit(tokenA.address), new BigNumber(1000001e18))
+      expect((await feeBurner.getRouterForToken(tokenA.address)).toString()).to.equal(
+        [uniswap.address].toString()
+      )
+      expect(await feeBurner.routerAddresses(tokenA.address, 0)).to.equal(uniswap.address)
+    })
+
+    it('should have set the owner', async () => {
+      const owner: string = await feeBurner.owner()
+      assert.equal(owner, accounts[0])
+    })
+
+    it('should not be callable again', async () => {
+      await assertRevert(feeBurner.initialize(registry.address, [], [], [], [], { from: user }))
+    })
+  })
+
+  describe('#removeRouter()', () => {
+    it('removes a token', async () => {
+      await feeBurner.removeRouter(tokenA.address, uniswap.address, 0)
+      expect((await feeBurner.getRouterForToken(tokenA.address)).toString()).to.equal([].toString())
+
+      // Todo check the list here
+      // TODO check with more items
+    })
+  })
+
+  describe('#setDailyBurnLimit()', () => {
+    it('should only be called by owner', async () => {
+      await assertRevert(
+        feeBurner.setDailyBurnLimit(stableToken.address, goldAmountForRate, { from: user })
+      )
+    })
+  })
+
+  describe('#setMaxSplipagge()', () => {
+    it('should only be called by owner', async () => {
+      await assertRevert(
+        feeBurner.setMaxSplippage(stableToken.address, maxSlippage, { from: user })
+      )
+    })
+  })
+
+  describe('#burnMentoTokens()', () => {
+    beforeEach(async () => {
+      const goldTokenAmount = new BigNumber(1e18)
+
+      await goldToken.approve(exchange.address, goldTokenAmount, { from: user })
+      await exchange.sell(goldTokenAmount, 0, true, { from: user })
+    })
+
+    it("Can't burn when forzen", async () => {
+      await freezer.freeze(feeBurner.address)
+      await assertRevert(feeBurner.burnMentoTokens())
+    })
+
+    it('burns with balance', async () => {
```

Comment:
this is a bit unclear.

```suggestion
    it('burns contract balance', async () => {
```